### PR TITLE
sql: guard purification of experimental features

### PR DESCRIPTION
**Pull request by petrosagg**

### CODE REVIEW

File: src/sql/src/pure.rs [hunk 1]
```diff
@@ -47,6 +48,7 @@ use crate::normalize;
 /// [`Catalog`](crate::catalog::Catalog), as that would require locking access
 /// to the catalog for an unbounded amount of time.
 pub fn purify(
+    scx: &StatementContext,
```
Comment (benesch) [inline src/sql/src/pure.rs:51]: So passing a `StatementContext` to purify is not the intended use of `StatementContext`. (Weird to pass in param types, and it duplicates the catalog). I think just check `catalog.config().experimental_mode` in the if?

**Review by benesch — CHANGES_REQUESTED** [review]

File: src/sql/src/pure.rs (hunk 1 above)
Comment (petrosagg) [reply src/sql/src/pure.rs]: I changed it to use the catalog directly. For my education, what is the intended use of the various context objects?

**Review by benesch — APPROVED** [review]
Looks great, thank you!

You can think of `StatementContext` as roughly "the specific stuff you need to pass to `plan_statement`. It's meant to be a single purpose type for that one public function. (The reason it's a struct rather than individual options is because internally in the planner it gets threaded through a bunch of things.)

**Comment by benesch** [comment]
I fixed up the merge conflict and marked for auto merge!

**Pull request merged**